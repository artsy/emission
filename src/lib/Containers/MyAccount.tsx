import * as React from "react"
import * as Relay from "react-relay"

import { Dimensions, ScrollView, View, ViewProperties, ViewStyle } from "react-native"

import Headline from "../components/text/headline"

interface Props extends ViewProperties, RelayProps {}

export class MyAccount extends React.Component<Props, {}> {
  state: {
    selectedTabIndex: number
  }

  render() {
    const windowDimensions = Dimensions.get("window")
    const commonPadding = windowDimensions.width > 700 ? 40 : 20

    return (
      <ScrollView scrollsToTop={true} automaticallyAdjustContentInsets={false}>
        <View style={{ paddingLeft: commonPadding, paddingRight: commonPadding }}>
          <Headline>{this.props.me.name}</Headline>
        </View>
      </ScrollView>
    )
  }
}

interface Styles {
  tabView: ViewStyle
}

export default Relay.createContainer(MyAccount, {
  fragments: {
    me: () => Relay.QL`
      fragment on Me {
        name
      }
    `,
  },
})

interface RelayProps {
  me: {
    name: string | null
  }
}
