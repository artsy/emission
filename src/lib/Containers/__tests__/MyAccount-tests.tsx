import * as React from "react"
import * as renderer from "react-test-renderer"

import { MyAccount } from "../MyAccount"

it("looks like expected", () => {
  const props = {
    me: {
      name: "Danger McShane",
    },
  }
  const tree = renderer.create(<MyAccount me={props.me} />).toJSON()
  expect(tree).toMatchSnapshot()
})
