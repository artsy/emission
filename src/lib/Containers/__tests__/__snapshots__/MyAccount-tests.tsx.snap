// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`looks like expected 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  scrollsToTop={true}
>
  <View>
    <View
      style={
        Object {
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontSize": 12,
            },
            undefined,
            Object {
              "fontFamily": "Avant Garde Gothic ITCW01Dm",
            },
          ]
        }
      >
        DANGER MCSHANE
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
